### PR TITLE
[SYCL][UR] Combined CTS fixes for CL adapter.

### PR DESCRIPTION
**Pull request by aarongreig**
Pulls in fixes from https://github.com/oneapi-src/unified-runtime/pull/1044

### CODE REVIEW

File: sycl/plugins/unified_runtime/CMakeLists.txt [hunk 1]
```diff
@@ -53,14 +53,17 @@ endif()
...
+  # commit f65473d9315c1319538f481d7a8c82dd2710c933
+  # Author: Aaron Greig <aaron.greig@codeplay.com>
+  # Date:   Mon Nov 6 16:35:33 2023 +0000
+  #
+  # [OpenCL] Add bounds checking to the Enqueue memory operations.
+  #
+  # This allows us to return UR_ERROR_INVALID_SIZE when we should. Extra
+  # checks are only performed on a non-success error code.
+  # Also adds a missing bounds check to urMemBufferPartition
+  set(UNIFIED_RUNTIME_TAG f65473d9315c1319538f481d7a8c82dd2710c933)
```
Comment (kbenzie) [inline sycl/plugins/unified_runtime/CMakeLists.txt:66]:
```suggestion
  set(UNIFIED_RUNTIME_REPO "https://github.com/oneapi-src/unified-runtime.git")
  # commit 192e9404392c38ac43d09116d6c97e153c66f46b
  # Merge: 2f44433c f65473d9
  # Author: Kenneth Benzie (Benie) <k.benzie@codeplay.com>
  # Date:   Fri Nov 10 13:25:42 2023 +0000
  #     Merge pull request #1044 from aarongreig/aaron/clCTSFixMegaBranch
  #     [OpenCL] Combined CTS fixes
  set(UNIFIED_RUNTIME_TAG 192e9404392c38ac43d09116d6c97e153c66f46b)
```

**Review by kbenzie — CHANGES_REQUESTED** [review]
Not that https://github.com/oneapi-src/unified-runtime/pull/1044 has merged, please pull in the latest sycl branch changes and update the `UNIFIED_RUNTIME_REPO`/`UNIFIED_RUNTIME_TAG` values as suggested. Then make this PR ready to review.

**Review by kbenzie — APPROVED** [review]
UR LGTM

**Review by martygrant — APPROVED** [review]

**Comment by kbenzie** [comment]
@intel/llvm-reviewers-runtime please review, it is intended for the next release.

File: sycl/plugins/unified_runtime/pi2ur.hpp [hunk 2]
```diff
@@ -2336,8 +2336,18 @@ inline pi_result piKernelGetInfo(pi_kernel Kernel, pi_kernel_info ParamName,
     break;
   }
   case PI_KERNEL_INFO_NUM_ARGS: {
-    UrParamName = UR_KERNEL_INFO_NUM_ARGS;
-    break;
+    size_t NumArgs = 0;
```
[the remark this answers is not in the document]
Comment (aarongreig) [reply sycl/plugins/unified_runtime/pi2ur.hpp:2339]: It turned out this particular query has always been `size_t` in UR, one of the fixes we're pulling in causes that to be properly validated now so we need to do this bit of translation. We do plan to fix this for the next release cycle https://github.com/oneapi-src/unified-runtime/issues/1038

**Review by aelovikov-intel — APPROVED** [review]

**Pull request merged**